paws/values.yaml

```diff
@@ -20,126 +66,172 @@ jupyterhub:
-                        'name': 'dumps',
-                        'hostPath': { 'path': '/public/dumps' }
-                    },
-                    {
-                        'name': 'dumps-mnt',
-                        'hostPath': { 'path': dumps_mnt }
-                    }
-                ]
-                spawner.volume_mounts = [
-                    {
-                        'name': 'home',
-                        'mountPath': '/home/paws'
-                    },
-                    {
-                        'name': 'dumps',
-                        'mountPath': '/public/dumps'
-                    },
-                    {
-                        'name': 'dumps-mnt',
-                        'mountPath': dumps_mnt
-                    }
-                ]
-        
-                spawner.environment['MYSQL_HOST'] = os.environ['MYSQL_SERVICE_HOST']
-                mysql_password = hmac.new(
-                    os.environ['MYSQL_HMAC_KEY'].encode('utf-8'),
-                    identity['username'].encode('utf-8'),
-                    hashlib.sha256
-                ).hexdigest()
-                spawner.environment['MYSQL_USERNAME'] = identity['username']
-                spawner.environment['MYSQL_PASSWORD'] = mysql_password
-        c.KubeSpawner.singleuser_fs_gid = 52771
-        
-        c.JupyterHub.authenticator_class = Auth
-        c.JupyterHub.logo_file = '/srv/jupyterhub/PAWS.svg'
-        c.JupyterHub.template_vars = {
-            'announcement': ('<span class="alert-success">'
-                             'Welcome to PAWS. '
-                             'Please <a href="//phabricator.wikimedia.org/tag'
-                             '/paws/"> report any issues on Phabricator </a>'
-                             '</span>')}
+      readOnly: true
+    - name: dumps-src1
+      mountPath: /mnt/nfs/dumps-labstore1007.wikimedia.org
+      readOnly: true
+    - name: dumps-src2
+      mountPath: /mnt/nfs/dumps-labstore1006.wikimedia.org
+      readOnly: true
+    extraConfig: 
+      myConfig: |
+          import hmac
+          import hashlib
+          import subprocess
+          import os
+          from oauthenticator.mediawiki import MWOAuthenticator
+          from tornado import gen
+          
+          class Auth(MWOAuthenticator):
+              enable_auth_state = True
+              def normalize_username(self, username):
+                  return username
+          
+              @gen.coroutine
+              def pre_spawn_start(self, user, spawner):
+                  auth_state = yield user.get_auth_state()
+                  identity = auth_state['MEDIAWIKI_USER_IDENTITY']
+                  spawner.environment['ACCESS_KEY'] = auth_state['ACCESS_TOKEN_KEY']
+                  spawner.environment['ACCESS_SECRET'] = auth_state['ACCESS_TOKEN_SECRET']
+                  spawner.environment['CLIENT_ID'] = self.client_id
+                  spawner.environment['CLIENT_SECRET'] = self.client_secret
```

Comment:
The OAuth magic for pywikibot! I think it would be nice to have a comment here explaining why this is done. Maybe something like:

    # Populate environment with OAuth grant data for use by pywikibot
    # See paws/images/singleuser/user-config.py for usage

Reply:
Definitely needed, but I'm not sure that needs to be in this already omnibus-level PR, though. Maybe a smaller PR that's all about documenting this file?
Helm best practice (that I'm not sure I've seen people follow in the wild) says that every line of the values.yaml should have a comment explaining the value's purpose and function, starting with the name of the value. This block of python overriding a value in jupyterhub's upstream chart, `ensures that the spawner sets up your CloudVPS environment for NFS usage and enables your OAuth for PWB and all access to PAWS, period`.  A comment like that as well as some inline comments like you said might make it a lot easier to hack on.  This PR doesn't add the segment, though, it just shifts it to the right a little to keep up with the upstream chart's deprecations and requirements.